projects/js-packages/publicize-components/src/components/share-status/modal-trigger.tsx

```diff
@@ -16,8 +16,13 @@ type ModalTriggerProps = ButtonProps & {
 export const ModalTrigger = forwardRef(
 	( { withWrapper = false, ...props }: ModalTriggerProps, ref: unknown ) => {
 		const { openShareStatusModal } = useDispatch( socialStore );
-
 		const featureFlags = useSelect( select => select( socialStore ).featureFlags(), [] );
+		const shareStatus = useSelect( select => select( socialStore ).getPostShareStatus(), [] );
```

Comment:
We need to check for feature flag here

```ts
const shareStatus = useSelect(
	select => {
		return featureFlags.useShareStatus
			? select( socialStore ).getPostShareStatus()
			: { shares: [] };
	},
	[ featureFlags.useShareStatus ]
);

// If the post is not shared anywhere, thus there are no shares, we don't need to show the trigger.
if ( ! featureFlags.useShareStatus || ! shareStatus.shares.length ) {
	return null;
}
```

Reply:
Do we do that here, or inside the `getPostShareStatus` resolver?

Reply:
I thought we could do something like 
```
		const featureFlags = registry.select( socialStore ).featureFlags();

		if ( ! featureFlags.useShareStatus ) {
			dispatch( fetchPostShareStatus( postId, false ) );
			return;
		}
```
into getPostShareStatus in `projects/js-packages/publicize-components/src/social-store/resolvers.js` so it's checked wherever its called

Reply:
I had thought the same but it's better to keep the resolver clean.

Reply:
Or may be it's fine to do that.

Reply:
Yeah that's true, but then we have to do that logic you mentioned everywhere 🤔 I'm not sure which one is better, I thought in the resolver it's less places to do

Reply:
OK, let us go with the resolver approach to do an early return based on feature flag